crates/sui-json-rpc/src/read_api.rs

```diff
@@ -678,8 +679,16 @@ impl ReadApiServer for ReadApi {
 
         let normalized = match object_read {
             ObjectRead::Exists(_obj_ref, object, _layout) => match object.data {
-                Data::Package(p) => normalize_modules(p.serialized_module_map().values())
-                    .map_err(|e| anyhow!("{e}")),
+                Data::Package(p) => {
+                    // TODO: is it OK to use it here since it's not on the actual validator?
```

Comment:
Good question -- it may not be necessary?   If we are just reading data from the store, it might be sufficient to use the max version in the binary.

Reply:
Yes, but given that this is (i think?) running in a fullnode, we get the current version from the fullnode with `self.state.epoch_store_one_call_per_task().protocol_config()`

Does it matter that we might be loading a package that was published in an earlier epoch / earlier protocol version?

Reply:
> Does it matter that we might be loading a package that was published in an earlier epoch / earlier protocol version?

I'd say it is better to use a more precise info on the max Move binary version since it's available (was not sure how to get to it - thanks @mystenmark!)

Reply:
Incidentally, this is one of the cases where if you use the max bytecode version from the protocol config as part of the  read API, we will run into problems if `ChangeEpoch` writes out a package with a newer bytecode version than the max for the current epoch's protocol.

FWIW, I think using the protocol config max bytecode format version is an unnecessary complication on the read path -- it's work that we don't need to do, and we can also end up tripping ourselves up on it later.